src/components/shared/form/createProject.tsx

```diff
@@ -0,0 +1,291 @@
+
+  const styles = {
+    control: (styles: any) => {
+      return {
+        ...styles,
+        border: formErrors.includes('pTech')
+          ? '1px solid red'
+          : '1px solid lightgray;',
+        background: formErrors.includes('pTech') ? '#fff1f4' : 'white',
+      };
+    },
+    multiValue: (styles: any) => {
+      return {
+        ...styles,
+        backgroundColor: '#5f8ddc',
+      };
+    },
+    multiValueLabel: (styles: any) => ({
+      ...styles,
+      color: 'white',
+    }),
+    multiValueRemove: (styles: any) => ({
+      ...styles,
+      color: 'white',
+      ':hover': {
+        backgroundColor: '#486ca8',
+        color: 'white',
+      },
+    }),
+  };
+
+  useEffect(() => {
+    async function fetchProjectTypes() {
+      const projTypes: any = await api.getProjectTypes();
+      setProjectTypes([...projTypes.data]);
+    }
+
+    fetchProjectTypes();
+  }, []);
+
+  const promiseOptions = async (inputValue: string) => {
+    const json = await fetch(
+      `https://api.stackexchange.com/2.2/tags?site=stackoverflow&key=*08t5pMLzA0X50xU9dNGbQ((&inname=${inputValue}`,
+    );
+    const data = await json.json();
+    return data.items.map((tag: { name: string }) => ({
+      value: tag.name,
+      label: tag.name,
+    }));
+  };
+
+  const handleSelectChange = (e: any) => {
+    let technologies = formInputs.pTech;
+    technologies = e ? e.map((tag: any) => ({ name: tag.value })) : [];
+    setFormInputs({
+      ...formInputs,
+      pTech: { ...formInputs.pTech, val: technologies },
+    });
+  };
+
+  const handleChange = (e: any, val = '') => {
+    const { name, value } = e.target;
+    const state: any = formInputs;
+    state[name].val = value;
+
+    if (name === 'pDesc') state[name].val = val;
+
+    setFormInputs({ ...state });
+  };
+
+  const handleBlur = (e: any) => {
+    const { name } = e.target;
+    const formErrorState: any = formErrors;
+    const formInputState: any = formInputs;
+
+    if (formInputState[name].val) {
+      formErrorState.splice(formErrorState.indexOf(name), 1);
+    } else {
+      formErrorState.push(name);
+    }
+
+    setFormErrors([...formErrorState]);
+  };
+
+  const onInputChange = (e: any) => {
+    const formErrorState: any = formErrors;
+
+    if (formErrorState.indexOf('pTech') > 0 || formInputs.pTech.val.length) {
+      if (formInputs.pTech.val.length) {
+        formErrorState.splice(formErrorState.indexOf('pTech'), 1);
+      }
+    } else {
+      if (!formInputs.pTech.val.length) {
+        formErrorState.push('pTech');
+      }
+    }
+
+    setFormErrors([...formErrorState]);
+  };
+
+  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    const { pName, pDesc, pTech, pType, pRepo, pLaunch, pComm } = formInputs;
+    const errors = formValidation(formInputs);
+
+    if (errors.length) return setFormErrors([...errors]);
+
+    const formData = {
```

Comment:
you're missing some properties that need to be sent to the api. To avoid missing properties - import the Project interface here and make `formData` of type Project.

```suggestion
    const formData: Project = {
```
Then update formData the missing properties.

Reply:
fixed